silence/silence.go

```diff
@@ -425,7 +479,7 @@ type query struct {
 
 // silenceFilter is a function that returns true if a silence
 // should be dropped from a result set for a given time.
-type silenceFilter func(*pb.Silence, *timestamp.Timestamp) (bool, error)
+type silenceFilter func(*pb.Silence, matcherCache, *timestamp.Timestamp) (bool, error)
```

Comment:
For being generic, let's pass in the entire `Silences` object here. Who knows what other caches or information we want to extend this by.